client/dashboard/domains/forwarding/index.tsx

```diff
@@ -0,0 +1,193 @@
+import { useSuspenseQuery, useMutation } from '@tanstack/react-query';
+import { Link, useRouter } from '@tanstack/react-router';
+import { Button } from '@wordpress/components';
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo } from 'react';
+import {
+	domainForwardingDeleteMutation,
+	domainForwardingQuery,
+} from '../../app/queries/domain-forwarding';
+import { domainRoute, domainForwardingAddRoute, domainForwardingEditRoute } from '../../app/router';
+import DataViewsCard from '../../components/dataviews-card';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import type { DomainForwardingObject } from '../../data/domain-forwarding';
+import type { Action, Field, ViewTable, ViewList, View } from '@wordpress/dataviews';
+
+function getForwardingId( forwarding: DomainForwardingObject ) {
+	return `${ forwarding.domain_redirect_id }-${ forwarding.domain }`;
+}
+
+type ForwardingView = ViewTable | ViewList;
+
+const DEFAULT_VIEW: ForwardingView = {
+	type: 'table',
+	search: '',
+	page: 1,
+	perPage: 20,
+	titleField: 'source',
+	sort: {
+		field: 'source',
+		direction: 'asc',
+	},
+	fields: [ 'destination' ],
+	filters: [],
+};
+
+const DEFAULT_LAYOUTS = {
+	table: {},
+	list: {},
+};
+
+function DomainForwarding() {
+	const router = useRouter();
+
+	const { domainName } = domainRoute.useParams();
+	const { data: forwardingData, isLoading } = useSuspenseQuery(
```

Comment:
Since data might not be ready (as we didn't `await` for it in the loader), we should use plain `useQuery()` instead.

```suggestion
	const { data: forwardingData, isLoading } = useQuery(
```